sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -529,69 +603,178 @@ myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
+descriptors that can be queried using `get_info` API.
 
-### Memory scope
-The current experimental API uses `joint_` semantics to define the memory scope of the matrix. The long term solution is to use the proposed link:../supported/sycl_ext_oneapi_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
+[frame="none",options="header"]
+|======================
+| Device descriptors | Return type| Description
+|`ext::oneapi::experimental::info::device::matrix::combinations` |
+`std::vector<combination>`| tells the set of supported matrix sizes
+and types on this device
+|======================
 
+The general query returns a vector of `combinations` of `combination`
+type. Each combination includes the sizes and the types for the
+matrices A, B, and accumulator. Note that for each matrix hardware,
+the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize,
+ksize` exclusively, depending on whether the implementation supports a
+continuous or discrete number of sizes. If a device support a
+continuous number of sizes, the `max_*` variant is applied and only
+the maximum number is returned. However, if a device supports a
+discrete list of numbers so the `msize, nsize, ksize` variant is applied.
 
 ```c++
-multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN, use::a>>(sg);
-```
-We did not utilize this extension for this matrix API version because sub-group local memory is not yet well defined in {dpcpp}. Moreover, the representation of this notion in LLVM IR and SPIR-V is not clear yet. 
+namespace sycl::ext::oneapi::experimental::matrix {
 
-### WI data to joint matrix mapping coordinates information for piece-wise operations
-The indexing provided inside the `wi_data` class accesses only the portion of the matrix held by the current WI. It is not possible to know the location of this portion in the original matrix.  This coordinates mapping  is implementation defined and changes from one backend to the other. For general piece-wise operations like sum of rows of a matrix, the WI data to joint matrix mapping information is needed to reason about the matrix view.
-Within the joint matrix extension, we want to write, as much as possible, one code to run on different backends. If backend X states that a WI owns one exact row of the matrix for instance, writing the following code will work only on that backend for that version of hardware. If a different hardware and implementation is used, the same WI may own only half of the row if, for example, the SG size increased. 
+enum class matrix_type {
+  bf16,
+  fp16,
+  tf32,
+  fp32,
+  fp64,
+  sint8,
+  sint16,
+  sint32,
+  sint64,
+  uint8,
+  uint16,
+  uint32,
+  uint64
+};
+struct combination {
+  uint32_t max_msize;
+  uint32_t max_nsize;
+  uint32_t max_ksize;
+  uint32_t msize;
+  uint32_t nsize;
+  uint32_t ksize;
```

Comment:
These should all be `size_t` since the template parameters for `Rows` and `Cols` are `size_t`.

Reply:
Ping on this comment.